ideascube/mediacenter/management/commands/remove_medias.py

```diff
@@ -0,0 +1,92 @@
+# -*- coding: utf-8 -*-
+import os
+import sys
+import argparse
+from os.path import join as pj
+
+from django.conf import settings
+from django.core.management.base import BaseCommand
+
+from ideascube.mediacenter.models import Document
+from ideascube.utils import printerr
+
+
+class Command(BaseCommand):
+    help = ('Remove files from the mediacenter.')
+
+    def add_arguments(self, parser):
+        subs = parser.add_subparsers(
+            title='Commands', dest='cmd', metavar='',
+            parser_class=argparse.ArgumentParser)
+        remove = subs.add_parser(
+            'clean_orphans',
+            help='Clean mediacenter files not associated with a document.')
+        remove.set_defaults(func=self.clean_orphans)
+        parser.add_argument('--dry-run', action='store_true',
+                            help='Print the list of medias that would be '
+                                 'removed. Do not actually remove them')
+
+    def abort(self, msg):
+        self.stderr.write(msg)
+        sys.exit(1)
+
+    def handle(self, *args, **options):
+        if 'func' not in options:
+            self.parser.print_help()
+            self.parser.exit(1)
+
+        options['func'](options)
+
+    def clean_orphans(self, options):
+        files_to_remove = self._get_orphan_files()
+        if options['dry_run']:
+            print("Files to remove are :")
+            for _file in files_to_remove:
+                print(" - '{}'".format(_file))
+        else:
+            for f in files_to_remove:
+                try:
+                    os.unlink(f)
+                except Exception as e:
+                    printerr("ERROR while deleting {}".format(f))
+                    printerr("Exception is {}".format(e))
+
+    def _get_orphan_files(self):
+        # List all (original and preview) files in the fs
+        original_files_root_dir = pj(settings.MEDIA_ROOT,
+                                     'mediacenter/document')
+        try:
+            _, _, original_files_in_fs = next(os.walk(original_files_root_dir))
+        except StopIteration:
```

Comment:
> Cause I want the files only. Using the os.walk avoid me to filter the result of os.listdir to remove directories.

Oh right! That's what I had missed. Indeed!

> Either I assume there is none of them and I can use os.listdir without filtering the list or I try to be defensive and I have to do a filtering anyway.

Or, you could use `os.scandir()` which is optimized just for this case: `entry.is_file()` doesn't require an additional system call, contrarily to `os.path.is_file(path)`. But that's Python 3.5 😭

(interesting, I just mistyped `Python 3.5` by forgetting the `shift` key, which gave ` Python ";(`, which I found highly appropriate 😆)

Another option would be to just not care about directories. When you try removing the files, it would raise `IsADirectoryError` for these, and you can just safely ignore that error, because who cares about empty directories anyway. (and as you said, there just won't be any)